Detectors/TPC/monitor/macro/RunFindAdcError.C

```diff
@@ -0,0 +1,171 @@
+//#if (!defined(__CINT__) && !defined(__CLING__)) || defined(__MAKECINT__)
+#include "FairLogger.h"
+
+#include "TPCBase/PadPos.h"
+#include "TPCReconstruction/RawReader.h"
+
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <memory>
+#include <algorithm>
+#include <thread>
+#include <mutex>
+
+#include <boost/tokenizer.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include "TH2F.h"
+
+
+TH2F* hAdcError = 0x0;
+
+
+bool run_parallel = true;
+
+using namespace std;
+using namespace o2::TPC;
+//#endif
+
+struct Result {
+  int mRun;
+  int mEvent;
+  int mRegion;
+  int mLink;
+  int mSampa;
+  int mTimebin;
+  int mSyncPos;
+  
+  Result() : mRun(-1), mEvent(-1), mRegion(-1), mLink(-1), mSampa(-1), mTimebin(-1), mSyncPos(-1) {};
+  Result(int run, int event, int region, int link, int sampa, int timebin, int syncpos) :
+    mRun(run), mEvent(event), mRegion(region), mLink(link), mSampa(sampa), mTimebin(timebin), mSyncPos(syncpos) {};
+};
+
+void loopReader(std::shared_ptr<RawReader> reader_ptr, std::vector<Result>& result_ptr) {
+  auto reader = reader_ptr.get();
+  while (reader->loadNextEventNoWrap() >= 0) {
+    if (reader->getAdcError().size() != 0) {
+      for (auto err : reader->getAdcError()){
```

Comment:
consider using
const auto&
unless you need a copy of the object or does it return a pointer?